synapse/storage/databases/main/registration.py

```diff
@@ -360,6 +360,35 @@ def set_server_admin_txn(txn):
 
         await self.db_pool.runInteraction("set_server_admin", set_server_admin_txn)
 
+    async def set_shadow_banned(self, user: UserID, shadow_banned: bool) -> None:
+        """Sets whether a user shadow-banned.
+
+        Args:
+            user: user ID of the user to test
+            shadow_banned: true iff the user is to be shadow-banned, false otherwise.
+        """
+
+        def set_shadow_banned_txn(txn):
+            self.db_pool.simple_update_one_txn(
+                txn,
+                table="users",
+                keyvalues={"name": user.to_string()},
+                updatevalues={"shadow_banned": shadow_banned},
+            )
+            # In order for this to apply immediately, reset all access tokens of this user.
```

Comment:
I think you mean clear the cache? When I first read this I thought you meant deleting the access tokens

Reply:
Yeah I mean clear the cache. 👍 I'll update the wording.